### PR TITLE
Refactored Factory::getUser() to ::getApplication()->getIdentity()

### PR DESCRIPTION
**Pull request by pe7er**
Joomla 4.1 uses the deprecated ``Factory::getUser()``. I've changed it in a couple of files to ``Factory::getApplication()->getIdentity()``. 

- Does it make sense to manually replace it like this?
- Or will it be done automatically with a global search & replace?
- Please advice me if I could continue to refactor it like this in other files as well?

### Summary of Changes
This PR changes ``Factory::getUser`` in 5 files to ``Factory::getApplication()->getIdentity``.
In some cases I added Exception to the code block and as name space.

``$user   = Factory::getUser($userId);``
to
``$user   = Factory::getApplication()->getIdentity($userId);``

``$user = Factory::getUser();``
to
``$user   = Factory::getApplication()->getIdentity();``

``if (!Factory::getUser()->authorise('core.admin'))``
to
``if (!Factory::getApplication()->getIdentity()->authorise('core.admin'))``

### Testing Instructions

Please do a code inspection. 
It should not change any functionality and Joomla should continue work correctly.

### CODE REVIEW

File: administrator/components/com_actionlogs/src/Plugin/ActionLogPlugin.php [hunk 1]
```diff
@@ -50,18 +51,19 @@ abstract class ActionLogPlugin extends CMSPlugin
...
 	 * This method adds a record to #__action_logs contains (message_language_key, message, date, context, user)
 	 *
-	 * @param   array   $messages            The contents of the messages to be logged
-	 * @param   string  $messageLanguageKey  The language key of the message
-	 * @param   string  $context             The context of the content passed to the plugin
-	 * @param   int     $userId              ID of user perform the action, usually ID of current logged in user
+	 * @param array $messages The contents of the messages to be logged
+	 * @param string $messageLanguageKey The language key of the message
+	 * @param string $context The context of the content passed to the plugin
+	 * @param int $userId ID of user perform the action, usually ID of current logged in user
```
Comment (Quy) [inline administrator/components/com_actionlogs/src/Plugin/ActionLogPlugin.php:57]: Please revert.

**Comment by laoneo** [comment]
Thanks for your pr. Honestly I would like to see a change to introduce a user service which can be injected into the view, to get a user from rather exchanging a static function call with another.

**Comment by sandewt** [comment]
A small addition:

`Factory::getUser()->id`
or
`Factory::getUser()->get('id')`
to
`Factory::getApplication()->getIdentity()->id`

**Comment by sandewt** [comment]
>It should not change any functionality and Joomla should continue work correctly.

Point of interest:
[https://github.com/joomla/joomla-cms/pull/30834#issuecomment-1067956161](https://github.com/joomla/joomla-cms/pull/30834#issuecomment-1067956161) and
[https://github.com/joomla/joomla-cms/pull/30834#issuecomment-1070383253](https://github.com/joomla/joomla-cms/pull/30834#issuecomment-1070383253)

**Comment by wilsonge** [comment]
Generally speaking this is OK. Worth noting that behaviours aren't exactly identical as noted in https://github.com/joomla/joomla-cms/issues/36448 so you can't do a full auto-replace in an IDE.

@laoneo HTMLView's at least will also be hard coupled to the application for the active template (https://github.com/joomla/joomla-cms/blob/4.1-dev/libraries/src/MVC/View/HtmlView.php#L377). So unless we plan to start injecting that whilst obviously the statics aren't ideal - it might be that we inject the application into the view in the constructor going forwards anyhow. Even if we inject the user because there's going to be authentication checks throughout the view probably makes more sense in the constructor than as a service there anyhow.

**Comment by laoneo** [comment]
What about the [UserFactory](https://github.com/joomla/joomla-cms/blob/4.1-dev/libraries/src/User/UserFactoryInterface.php)?

**Comment by joomdonation** [comment]
@pe7er You might want to close this PR because we now use a different approach:

- PR https://github.com/joomla/joomla-cms/pull/37498 introduced a new way to get current user object in views and models class
- The Factory::getUser() replacement on view classes was merged https://github.com/joomla/joomla-cms/pull/38006
- There is also a PR made to replace Factory::getUser() in model classes, too, see https://github.com/joomla/joomla-cms/pull/38090

File: administrator/components/com_actionlogs/src/Model/ActionlogModel.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ class ActionlogModel extends BaseDatabaseModel
 	 */
 	public function addLog($messages, $messageLanguageKey, $context, $userId = null)
 	{
-		$user   = Factory::getUser($userId);
+		$user   = Factory::getApplication()->getIdentity($userId);
```
Comment (joomdonation) [inline administrator/components/com_actionlogs/src/Model/ActionlogModel.php:47]: Just for information the change here is not right. The getIdentity method does not accept any parameter like you passed here. If we want to replace it, we will need to use UserFactory, something like:

```php
$user = Factory::getContainer()->get(UserFactoryInterface::class)->loadUserById($userId);
```

**Comment by laoneo** [comment]
Agree, this one should be closed as users will be injected into views and models.

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Comment by pe7er** [comment]
@joomdonation thanks for your explanation and for your PR to solve this in a better way
@laoneo thanks for your PR to solve this in a better way

closing this one...

**Pull request closed without merging**